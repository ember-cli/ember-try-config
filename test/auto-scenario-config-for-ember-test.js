'use strict';

let autoScenarioConfigForEmber = require('../lib/auto-scenario-config-for-ember');

let fakeGetChannelUrl = function (channel) {
  return Promise.resolve(`https://emberjs.example.com/${channel}-1234.tgz`);
};

describe('lib/auto-scenario-config-for-ember', () => {
  test('includes default scenarios and works with straight version #', () => {
    jest.setTimeout(10000);

    return autoScenarioConfigForEmber({
      versionCompatibility: { ember: '2.18.0' },
      getChannelURL: fakeGetChannelUrl,
    }).then((config) => {
      expect(config.scenarios).toEqual([
        {
          name: 'default',
          npm: {
            devDependencies: {},
          },
        },
        {
          name: 'ember-beta',
          allowedToFail: true,
          npm: {
            devDependencies: {
              'ember-source': 'https://emberjs.example.com/beta-1234.tgz',
            },
            overrides: {
              'ember-source': '$ember-source',
            },
          },
        },
        {
          name: 'ember-canary',
          allowedToFail: true,
          npm: {
            devDependencies: {
              'ember-source': 'https://emberjs.example.com/canary-1234.tgz',
            },
            overrides: {
              'ember-source': '$ember-source',
            },
          },
        },
        {
          name: 'ember-release',
          npm: {
            devDependencies: {
              'ember-source': 'https://emberjs.example.com/release-1234.tgz',
            },
            overrides: {
              'ember-source': '$ember-source',
            },
          },
        },
        {
          name: 'ember-2.18.0',
          npm: {
            devDependencies: {
              'ember-source': '2.18.0',
            },
            overrides: {
              'ember-source': '$ember-source',
            },
          },
        },
      ]);
    });
  });

  test('works with complex semver statement', () => {
    let availableVersions = [
      '2.11.0',
      '2.12.0',
      '2.12.2',
      '2.13.0',
      '2.18.0',
      '3.0.0',
      '3.1.0',
      '3.1.1',
    ];

    return autoScenarioConfigForEmber({
      versionCompatibility: { ember: '2.11.0 - 2.12.7 || ~ 3.1.0 || ^2.18.0' },
      availableVersions,
      getChannelURL: fakeGetChannelUrl,
    }).then((config) => {
      expect(config.scenarios).toMatchSnapshot();
    });
  });
});
