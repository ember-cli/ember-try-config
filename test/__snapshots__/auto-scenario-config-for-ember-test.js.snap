// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lib/auto-scenario-config-for-ember works with complex semver statement 1`] = `
Array [
  Object {
    "name": "default",
    "npm": Object {
      "devDependencies": Object {},
    },
  },
  Object {
    "allowedToFail": true,
    "name": "ember-beta",
    "npm": Object {
      "devDependencies": Object {
        "ember-source": "https://emberjs.example.com/beta-1234.tgz",
      },
      "overrides": Object {
        "ember-source": "$ember-source",
      },
    },
  },
  Object {
    "allowedToFail": true,
    "name": "ember-canary",
    "npm": Object {
      "devDependencies": Object {
        "ember-source": "https://emberjs.example.com/canary-1234.tgz",
      },
      "overrides": Object {
        "ember-source": "$ember-source",
      },
    },
  },
  Object {
    "name": "ember-release",
    "npm": Object {
      "devDependencies": Object {
        "ember-source": "https://emberjs.example.com/release-1234.tgz",
      },
      "overrides": Object {
        "ember-source": "$ember-source",
      },
    },
  },
  Object {
    "name": "ember-2.11.0",
    "npm": Object {
      "devDependencies": Object {
        "ember-source": "2.11.0",
      },
      "overrides": Object {
        "ember-source": "$ember-source",
      },
    },
  },
  Object {
    "name": "ember-2.12.2",
    "npm": Object {
      "devDependencies": Object {
        "ember-source": "2.12.2",
      },
      "overrides": Object {
        "ember-source": "$ember-source",
      },
    },
  },
  Object {
    "name": "ember-2.18.0",
    "npm": Object {
      "devDependencies": Object {
        "ember-source": "2.18.0",
      },
      "overrides": Object {
        "ember-source": "$ember-source",
      },
    },
  },
  Object {
    "name": "ember-3.1.1",
    "npm": Object {
      "devDependencies": Object {
        "ember-source": "3.1.1",
      },
      "overrides": Object {
        "ember-source": "$ember-source",
      },
    },
  },
]
`;
